### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by dgageot**
+ Use the same bases images as much as possible
+ Simplify skaffold.yaml files
+ Shorten paths in images
+ Fix documentation

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: examples/environment-variables/skaffold.yaml [hunk 1]
```diff
@@ -1,13 +1,11 @@
 apiVersion: skaffold/v1alpha2
 kind: Config
 build:
+  artifacts:
+  - imageName: gcr.io/k8s-skaffold/skaffold-example
```
[the remark this answers is not in the document]
Comment (dgageot) [reply examples/environment-variables/skaffold.yaml]: The code does that already: https://github.com/GoogleContainerTools/skaffold/blob/78469db455903483c56423b5ce79c8a6065df476/pkg/skaffold/schema/v1alpha2/config.go#L200

[the remark this answers is not in the document]
Comment (r2d4) [reply examples/environment-variables/skaffold.yaml]: Oh I missed that, thanks!

**Comment by r2d4** [comment]
One test fail on TestEnvTagger

```
stderr: Error: running skaffold steps: deploy: deploy step: deploying manifests: running kubectl: stdout: , stderr: Error from server (NotFound): error when applying patch:
    		{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"v1\",\"kind\":\"Pod\",\"metadata\":{\"annotations\":{},\"name\":\"getting-started\",\"namespace\":\"default\"},\"spec\":{\"containers\":[{\"image\":\"gcr.io/k8s-skaffold/skaffold-example-foo\",\"imagePullPolicy\":\"IfNotPresent\",\"name\":\"getting-started\"}]}}\n"}},"spec":{"$setElementOrder/containers":[{"name":"getting-started"}],"containers":[{"image":"gcr.io/k8s-skaffold/skaffold-example-foo","name":"getting-started"}]}}
```

I think the image might be getting deleted from the previous integration runs. I'll add #419 which should be more correct

**Comment by r2d4** [comment]
ref #436 i think the example was broken to begin with

**Comment by r2d4** [comment]
one last run of ci

**Pull request merged**